cmake/Dependencies.cmake

```diff
@@ -154,3 +154,18 @@ list(APPEND DALI_EXCLUDES libprotobuf.a)
 # Exclude stdlib
 ##################################################################
 list(APPEND DALI_EXCLUDES libsupc++.a;libstdc++.a;libstdc++_nonshared.a;)
+
+##################################################################
+# TensorRT
+##################################################################
+if(BUILD_TENSORRT)
+    if(DEFINED ENV{TRT_RELEASE})
+        set(TRT_PATH $ENV{TRT_RELEASE})
+        include_directories(${TRT_PATH}/include)
+        include_directories(${TRT_PATH}/samples/common)
+        list(APPEND DALI_LIBS ${TRT_PATH}/lib/libnvinfer.so)
```

Comment:
So if you build with TENSORRT you need to have TRT installed to use DALI. If so it implies additional, silent dependency.
I would recommend to update dali/python/bundle-wheel.sh to include this *.so files as well.
Please also update Dockerfile.deps to install TENSORRT inside and enable BUILD_TENSORRT in the main Docker file so we can test it on our CI to see if that compiles and doesn't break anything.

Reply:
Not just TRT, but also cuDNN, cudart (to which TRT does not statically link), and cuBLAS.  There are huge ramifications here.

Reply:
For example, if we include this, it and if we bundle all those dependencies, the size of DALI will increase by over an order of magnitude.  If we don't bundle those dependencies, the portability of the built DALI to different systems without the same versions preinstalled will be negatively impacted.  You can't run a TRT built for CUDA 9 on a CUDA 10 system without also having cuBLAS 9 and cudart 9 installed, as TRT dynamically links both of those.

Reply:
@cliffwoolley  - that is right, I missed that. So we need to have a bigger discussion about how and if support this kind of use cases. As I saw TRT is the last operator in the pipeline - so maybe using SetCompletionCallback and running TRT in a separate thread will do and there is no need to create the operator for that.
Also, we may consider the ability to build plugins for DALI and distribute it separately - what is not the best idea and makes testing and maintainability more than difficult.

Reply:
Maybe we can think about operator that can call arbitrary external code. All this code need to do is accept workspace pointer and do the rest internally.
I think current operator schema enable us to be flexible regarding the number of operator inputs and outputs.

Reply:
I have taken this discussion offline on mail.